catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/event/impl/SubscriptionImpl.java

```diff
@@ -63,4 +65,15 @@ public DeliveryMethod getDeliveryMethod() {
         return dm;
     }
 
+    @Override
+    public boolean isActive() {
+        return active;
+    }
+
+    @Override
+    public boolean checkActive() {
+        active = dm.ping();
```

Comment:
Is the `ping()` and expensive operation? Will `isActive()` be called a large number of times vs `checkActive()`? I'm curious if it wouldn't be simpler/cleaner to remove one of these two methods from the interface and get rid of the field entirely, just returning `dm.ping()` on every call.